sdk/core/core-http/src/fetchHttpClient.ts

```diff
@@ -65,11 +65,12 @@ export abstract class FetchHttpClient implements HttpClient {
     if (httpRequest.formData) {
       const formData: any = httpRequest.formData;
       const requestForm = new FormData();
-      const appendFormValue = (key: string, value: any) => {
+      const appendFormValue = (key: string, value: any): void => {
         // value function probably returns a stream so we can provide a fresh stream on each retry
         if (typeof value === "function") {
           value = value();
         }
+        // eslint-disable-next-line no-prototype-builtins
         if (value && value.hasOwnProperty("value") && value.hasOwnProperty("options")) {
```

Comment:
maybe calling `hasOwnProperty` off the value received isn't a good idea?

Reply:
Should I do that fix as part of this PR?

Reply:
Log an issue for future follow up. It is preferable if the first pass at eslint focuses on simple, trivial fixes.

Reply:
I've made a new issue: https://github.com/Azure/azure-sdk-for-js/issues/8972

Reply:
@ramya-rao-a I've assigned #8972 to myself, but in case you know a better resource for that issue, feel free to re-assign it.

Reply:
Hmm. My intuition here is to say that calling `hasOwnProperty` should be fine in this case, since the formData will come from our clients. I suppose if someone imports core-http and calls sendRequest manually with a nefarious formData field, then it could trigger an exception. Probably no harm in defensively substituting `Object.prototype.hasOwnProperty.call(value, "value")`.

Reply:
from what I could tell looking through our repo, this weird form of passing options is never used. Also it doesn't make sense for the xhr client which only has an optional filename rather than an options bag like the node `form-data` module.

So at least for v2, I'm gonna drop this bit.